t/test.py

```diff
@@ -613,6 +613,42 @@ def test_timestamp_not_implemented(self):
         with self.assertRaises(NotImplementedError):
             jdt.timestamp()
 
+    def test_isoformat_default_args(self):
+        jdt = jdatetime.datetime(1398, 4, 11)
+        jiso = jdt.isoformat()
+
+        self.assertAlmostEqual(jiso, '1398-04-11T00:00:00')
+
+    def test_isoformat_custom_sep(self):
+        jdt = jdatetime.datetime(1398, 4, 11)
+        jiso = jdt.isoformat('M')
+
+        self.assertAlmostEqual(jiso, '1398-04-11M00:00:00')
+
+    def test_isoformat_bad_sep(self):
+        jdt = jdatetime.datetime(1398, 4, 11)
+
+        with self.assertRaises(AssertionError):
+            jdt.isoformat('dummy')
+            jdt.isoformat(123)
```

Comment:
Good idea to test for different invalid separators, however this statement and the others below won't execute, because the `jdt.isoformat('dummy')` raises.

Reply:
it's better now